www/activities/foods-meals-recipes/js/foods-meals-recipes.js

```diff
@@ -516,17 +516,21 @@ app.FoodsMealsRecipes = {
         let text = "";
 
         if (item.name != "Quick Add" && item.portion !== undefined && !isNaN(item.portion)) {
-          text = parseFloat(item.portion);
+          text = app.Utils.tidyNumber(parseFloat(item.portion));
 
           if (item.unit !== undefined) {
             if (app.standardUnits.includes(item.unit))
-              text += item.unit;
+              text += "&thinsp;" + item.unit;
             else
               text += " " + app.Utils.escapeHtml(item.unit);
           }
```

Comment:
@EmilJunker: Why not just add an optional "unit" parameter to `app.Utils.tidyNumber()` and do this formatting there? It otherwise seems to be a bit inconsistent again.

Reply:
Well, it's not that simple. In some languages, e.g. Hebrew, there are special localized versions of certain units. And everything needs to work with a right-to-left layout too. The case distinction for `standardUnits` is really just a thing in the food list to make sure an item with unit "slices" doesn't show up as "2slices" (or "2 slices" with a thin space which also look weird). Everywhere else, it always uses normal spaces. But I like your idea. I think we can indeed make this case distinction part of the tidyNumber function to apply it everywhere.

Reply:
You could even replace and initial hyphen with the proper HTML `&minus;`, if it's for labels only (not textboxes meant to be edited).